src/main/kotlin/org/rust/lang/core/psi/RustBinaryExprElementEx.kt

```diff
@@ -0,0 +1,12 @@
+package org.rust.lang.core.psi
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.RustTokenElementTypes.BINARY_OPS
+import org.rust.lang.core.psi.util.elementType
+
+val RustBinaryExprElement.operator: PsiElement
+    get() = node.findChildByType(BINARY_OPS)!!.psi // guaranteed to be not-null by parser; null == bug
```

Comment:
It's better to use `reuireNotNull(thing) { "why is this impossible" }` instead of comments, because one day the impossible will happen and it would be convenient to see this justification in the stack trace :)